docs/ecdsa.md

```diff
@@ -0,0 +1,89 @@
+# ECDSA signature format
+
+When ECDSA SECP256R1 (EC256) signature support was added to MCUboot, a
+shortcut was taken, and these signatures were padded to make them
+always a fixed length.  Unfortunately, this padding was done in a way
+that is not easily reversible.  Some crypto libraries are fairly
+strict about the formatting of the ECDSA signature (specifically, mbed
+TLS).  This currently means that the ECDSA SECP224R1 (EC) signature
+checking code will fail to boot about 1 out of every 256 images,
+because the signature itself will end in a 0x00 byte, and the code
+will remove too much data, invalidating the signature.
+
+There are a couple of ways to fix this:
+
+  1.  Use a reversible padding scheme.  This will work, but requires
+      at least one pad byte always be added (to set the length).  This
+      padding would be somewhat incompatible across versions (older
+      EC256 would work, newer mcuboot code would reject old
+      signatures.  EC code would only reliably work in the new
+      combination).
+
+  2.  Remove the padding entirely.  Depending on which tool, this will
+      require some rethinking of how TLV generation is implemented so
+      that the length does not need to be known until the signature is
+      generated.  These tools are all written in higher-level
+      languages and this change should not be difficult.
+
+      However, this will also break compatibility with older version,
+      significantly in that images generated with newer tools will not
+      work with older versions of MCUboot.
+
+This document proposes a multi-stage approach, to give a transition
+period.
+
+  - First, add a `--valid-ecdsa` argument to the sign command in
+    `imgtool.py`.  Without this, the images will be padded with the
+    existing scheme, and with the argument, the ecdsa will be encoded
+    without any padding.
+
+  - MCUboot will be modified to allow unpadded signatures right away.
+    The existing EC256 implementations will still work (with or
+    without padding), and the existing EC implementation will begin
+    accepting all signatures.
```

Comment:
```suggestion
    accepting padded and unpadded signatures.
```

Is that correct? I'm not sure. The existing wording makes it sound like it will always accept any signature, even if it's invalid, which I'm guessing is not what you meant.